### PR TITLE
Add CMake support and build script

### PR DESCRIPTION
**Pull request by oqu**
On mac os, ./build.sh can succesfully build this project.

Dependencies are cloned if not already available.

### CODE REVIEW

**Comment by oqu** [comment]
@magicrb, @androiddrew, you can try build.sh and let me know ?

**Comment by MagicRB** [comment]
Won't build in my machine, and also, `CINDER_PATH=${THIS_PATH}/cinder` should be `CINDER_PATH=${THIS_PATH}/Cinder`. The not building part, is a my machine specific thing, I think, I had this same issue with klipper, and it's a missing set of headers. Not your fault, it would be build on my laptop, I think. @androiddrew can verify my assumption

**Comment by oqu** [comment]
@MagicRB , thanks for the check. I modified the build script.
It looks like Mac OS does not care about the case.

**Comment by neuroprod** [comment]
Thank you guys!

**Comment by androiddrew** [comment]
Unfortunately, `./build.sh` hasn't worked out the gate. 

```
Cinder has not been configure, running cmake ...
./build.sh: line 36: cmake: command not found
```

**Comment by androiddrew** [comment]
Whoops thought I had cmake..`brew install` fixed it.

File: build.sh [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+THIS_PATH=$(pwd)
+CINDER_PATH=${THIS_PATH}/Cinder
+exitWithError() {
+  echo "Error:" "$@" >&2
+  exit 1
+}
+
+checkRequiredTool() {
+  for tool in git make cmake ; do
+    command -v git &> /dev/null || exitWithError "Missing required tool ${tool}"
```
Comment (oqu) [inline build.sh:15]: That should have been command -v "${tool}"

**Pull request merged**